### ISSUE TITLE
Default term should be checked by default

### ISSUE THREAD

**Issue by helgatheviking** [helgatheviking/Radio-Buttons-for-Taxonomies#62]
See: https://wordpress.org/support/topic/set-default-term/

**Comment by jaboi** [comment]
Prior to Gutenberg, I was able to set a default term (that worked with the OP's plugin Radio Buttons for Taxonomies) using **wp_set_object_terms** -- but this no longer works. If another (non-default) taxonomy is selected, the post is saved with two terms, the default and the one selected. This renders the desired single-select outcome of the radio button functionality useless. I don't think this is the plugin author's fault because her plugin works great off the shelf. The problem is when I want to designate a default taxonomy choice for posts. I suspect because of how Gutenberg saves/updates, a different hook needs to be used...?

**Comment by jaboi** [comment]
For some reason I had the final $append parameter set to 'true' which allowed a tag to be appended. (But the appending behavior only manifested after one of the recent updates to Radio Buttons for Taxonomies.) After changing that parameter to 'false', wp_set_object_terms works great. Here's my solution for generating a default tag:

`// ~~~~~~~ SET DEFAULT TAG TO 'news'

function set_default_status_term( $post_id ) {
    $current_post = get_post( $post_id );

    // This makes sure the taxonomy is only set when a new post is created
    if ( $current_post->post_date == $current_post->post_modified ) {
        wp_set_object_terms( $post_id, 'news', 'post_tag', false );
    }
}
add_action( 'save_post', 'set_default_status_term' );`

**Comment by jaboi** [comment]
I was wrong. The problem was not Gutenberg. I had been able to set a default term with Radio Buttons for Taxonomies with Gutenberg. It had to do with an update to the plugin sometime during 2019. Because my previous solution was undesirably saving two terms (the one selected and the default term), I had to add a check to see if any terms had been assigned to the post yet. Now I get my default term of 'news' only if another term has not been selected (I'm doing this for post tags):

```
// ~~~~~~~ SET DEFAULT TAG TO 'news'

function set_default_status_term( $post_id ) {
    $current_post = get_post( $post_id );
    $the_tags = get_the_tags( $post_id );

    // This makes sure the taxonomy is only set when a new post is created
    if ( $current_post->post_date == $current_post->post_modified && $the_tags == false ) {
        wp_set_object_terms( $post_id, 'news', 'post_tag' );
    }
}
add_action( 'save_post', 'set_default_status_term' );
```

**Comment by helgatheviking** [comment]
I think RB4T has always been trying to save the terms on [`save_post`](https://github.com/helgatheviking/Radio-Buttons-for-Taxonomies/blob/master/inc/class.WordPress_Radio_Taxonomy.php#L60). The 4th parameter for `wp_set_object_terms()` determines if the terms are appended or not, but that's false by default so 2 terms getting saved is surprising to me.

FYI, you can pass the `$post` object to `save_post` to avoid fetching the post again. 

```
function set_default_status_term( $post_id, $current_post ) {

    $the_tags = get_the_tags( $post_id );

    // This makes sure the taxonomy is only set when a new post is created
    if ( $current_post->post_date === $current_post->post_modified && $the_tags === false ) {
        wp_set_object_terms( $post_id, 'news', 'post_tag' );
    }
}
add_action( 'save_post', 'set_default_status_term', 20, 2 );
```

**Comment by jaboi** [comment]
My original function had the 4th parameter — it was set to 'true' and it worked for setting a default tag (but only if another tag was not selected). But after updating your plugin from 1.8.3 to 2.0.3, it did start appending the default term with a selected term. I thought that parameter was the problem, so I changed it to 'false'. This solved the problem when publishing a post, but when saving as a draft, it only saves the default term (even if you select a different term). Since the authors on our site do a lot with drafts and previews of drafts, that wouldn't work, so I had to add the check for any existing tags.

**Comment by helgatheviking** [comment]
WP 5.5 supports a default term for the taxonomy... 
https://core.trac.wordpress.org/ticket/43517

related to #72  though I haven't noticed that the default is checked by default in the classic editor without RB4T so unsure how to handle that. and still not able to devote a lot of time to this plugin.

**Comment by mdingena** [comment]
I'm using RB4T to enable radio buttons for WordPress Categories. I am experiencing a problem where a new Category is being created, called `1`. This happens only when I am creating a new Post. Steps to reproduce:

1. Set RB4T to enable for Categories.
1. Create new Post.
1. Type a title, click somewhere else. Now wait a while for auto-save to kick in.
1. Reload the page. You will be asked if you are sure because you have unsaved changes. Do it anyways.
1. The page reloads and a new Category `1` will have appeared and have been selected.

Expected results:
`Uncategorized` should have been selected instead of a new category being created.

**Comment by helgatheviking** [comment]
Appreciate the report @mdingena If you've got any time to help dig in to this that will go a long way towards getting this fixed. not sure when i'll be able to look at this. and still not sure how the new default term should be handled at all.

**Comment by mdingena** [comment]
I've stopped using this plugin because I've taken a different approach to my problem.

However, in trying to solve the issue I stumbled onto [`get_sample_permalink`](https://developer.wordpress.org/reference/functions/get_sample_permalink/), which seems to return "`1`", but the documentation is not clear to me when this is the case.

Does your plugin use this function? It could also just be default WordPress behaviour, although I've never seen a `1` category be created on saving a draft before.
